test/test_reductions.py

```diff
@@ -504,6 +504,109 @@ def test_logsumexp(self, device):
         self.assertEqual(expected.shape, actual.shape)
         self.assertEqual(expected, actual)
 
+    @skipIfNoSciPy
+    @dtypes(torch.complex64, torch.complex128)
+    def test_logcumsumexp_complex(self, device, dtype):
+        # logcumsumexp is a more precise way to compute than ``log(cumsum(exp(a)))``
+        # and faster than ``[log(sum(exp(a[:i]))) for i in range(a.shape[0])]``
+        # the for-loop above should produce similar precision as logcumsumexp (it's just slower),
+        # so it can be used as the expected values to check our computation
```

Comment:
You might want to define this as a ref for the OpInfo and it will check that the ref match for all simple inputs without the need for this custom code. Or there are more cases tested here that cannot be tested via the OpInfo?

Reply:
I found several challenges in using `OpInfo` (I probably don't understand OpInfo enough to fully utliize it) for this complex logcumsumexp function:

1) To compare 2 outputs of `logcumsumexp` functions, the imaginary part needs to be standardized to be within `(0, 2 pi)` or `(-pi, pi)`. This is because `log(r*e^{i t}) = log|r| + i * (t + 2 pi * n)`.
2) scipy.logsumexp gives some confusing answers involving `inf` case (see my comment around line 590 of this file)